drivers/linux_platform/platform_drivers.c

```diff
@@ -192,20 +193,23 @@ int32_t spi_init(spi_desc **desc, const spi_init_param *param)
 	descriptor->fd = open(param->pathname, O_RDWR);
 	if (descriptor->fd < 0) {
 		printf("%s: Can't open device\n\r", __func__);
+		free(descriptor);
 		return FAILURE;
 	}
 
 	ret = ioctl(descriptor->fd, SPI_IOC_WR_MODE,
 			&param->mode);
 	if (ret == -1) {
 		printf("%s: Can't set mode\n\r", __func__);
+		free(descriptor);
```

Comment:
i think here you may also need to close the `fd` before free-ing the descriptor
so,
```
close(descriptor->fd);
free(descriptor);
```
similarly, this should be done at lines 212 & 300